jems-testing/src/main/java/org/dmfs/jems/hamcrest/matchers/AbsentMatcher.java

```diff
@@ -60,7 +64,18 @@ protected boolean matchesSafely(Optional<T> item, Description mismatchDescriptio
             // pass
         }
 
-        if (((Optional) item).value(DUMMY_DEFAULT) != DUMMY_DEFAULT)
+        T fallbackDummy = mFallbackDummySingle.value();
+        T fallbackMethodResult;
+        try
+        {
+            fallbackMethodResult = item.value(fallbackDummy);
+        }
+        catch (ClassCastException e)
+        {
+            throw new RuntimeException("ClassCastException in AbsentMatcher, need to use #absent(Class<T> c) or #absent(T t) method", e);
+        }
+
+        if (fallbackMethodResult != fallbackDummy)
```

Comment:
Lines 68, 71 and 78 should be merged into one. Splitting this expression over 3 lines doesn't make it any easier to read. 
This makes much more sense to me:

```
 T fallbackDummy = mFallbackDummySingle.value();
 if (item.value(fallbackDummy) != fallbackDummy)
```